apps/condo/domains/billing/access/BillingIntegrationOrganizationContext.js

```diff
@@ -7,42 +7,51 @@ const { get } = require('lodash')
 const { getById } = require('@core/keystone/schema')
 const { checkOrganizationPermission } = require('@condo/domains/organization/utils/accessSchema')
 const { throwAuthenticationError } = require('@condo/domains/common/utils/apolloErrorFormatter')
+const { USER_SCHEMA_NAME } = require('@condo/domains/common/constants/utils')
 
-async function canReadBillingIntegrationOrganizationContexts ({ authentication: { item: user } }) {
-    if (!user) return throwAuthenticationError()
-    if (user.isAdmin || user.isSupport) return true
-    return {
-        OR: [
-            { organization: { employees_some: { user: { id: user.id }, role: { OR: [{ canReadBillingReceipts: true }, { canManageIntegrations: true }] }, isBlocked: false } } },
-            { integration: { accessRights_some: { user: { id: user.id } } } },
-        ],
+async function canReadBillingIntegrationOrganizationContexts ({ authentication: { item, listKey } }) {
+    if (!listKey || !item) return throwAuthenticationError()
+    if (item.deletedAt) return false
+    if (listKey === USER_SCHEMA_NAME) {
+        if (item.isSupport || item.isAdmin) return true
+        return {
+            OR: [
+                { organization: { employees_some: { user: { id: item.id }, role: { OR: [{ canReadBillingReceipts: true }, { canManageIntegrations: true }] }, isBlocked: false } } },
+                { integration: { accessRights_some: { user: { id: item.id }, deletedAt: null } } },
+            ],
+        }
     }
+    return false
 }
 
-async function canManageBillingIntegrationOrganizationContexts ({ authentication: { item: user }, originalInput, operation, itemId, context }) {
-    if (!user) return throwAuthenticationError()
-    if (user.isAdmin) return true
-    if (user.isSupport && !get(originalInput, ['organization', 'connect', 'id'])) return true
-    let organizationId
-    let integrationId
-    if (operation === 'create') {
-        // NOTE: can only be created by the organization integration manager
-        organizationId = get(originalInput, ['organization', 'connect', 'id'])
-        integrationId = get(originalInput, ['integration', 'connect', 'id'])
+async function canManageBillingIntegrationOrganizationContexts ({ authentication: { item, listKey }, originalInput, operation, itemId }) {
+    if (!listKey || !item) return throwAuthenticationError()
+    if (item.deletedAt) return false
+    if (listKey === USER_SCHEMA_NAME) {
+        if (item.isAdmin) return true
+        if (item.isSupport && !get(originalInput, ['organization', 'connect', 'id'])) return true
+        let organizationId
+        let integrationId
+        if (operation === 'create') {
```

Comment:
this way it would be a bit more readable
```
        let organizationId, integrationId

        if (item.isAdmin) return true
        if (item.isSupport && !get(originalInput, ['organization', 'connect', 'id'])) return true
        if (operation === 'create') {
```